tests/scripts/rm_ml_api_tests.sh

```diff
@@ -0,0 +1,293 @@
+		
+		case "${post}" in
+			invalid-searchspace)
+				exp="valid-experiment"
+				sed 's/"sla_class": "response_time"/"sla_class": "xyz"/g' ${SEARCH_SPACE_JSON} > ${TESTS_}/invalid_searchspace.json
+				echo "Searchspace JSON" | tee -a ${LOG_} ${LOG}
+				cat ${TESTS_}/invalid_searchspace.json | tee -a ${LOG_} ${LOG}
+				echo "" | tee -a ${LOG_} ${LOG}
+				# Start the python servers
+				${SCRIPTS_DIR}/start_python_servers.sh -p ${TESTS_} -j ${TESTS_}/invalid_searchspace.json | tee -a ${LOG_} ${LOG}
+				;;
+			*)
+				exp="${post}"	
+				# Start the python servers
+				${SCRIPTS_DIR}/start_python_servers.sh -p ${TESTS_} -j ${SEARCH_SPACE_JSON} | tee -a ${LOG_} ${LOG}
+		esac
+		
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+		
+		# Get the id from search space JSON
+		current_id=$(cat ${SEARCH_SPACE_JSON} | jq .[].id | tr -d '""')
+		
+		get_json_array ${current_id}
+		post_json "${rm_ml_post_experiment_json[$exp]}"
+	
+		if [[ "${post}" == valid* ]]; then
+			expected_result_="200"
+			expected_behaviour="RESPONSE_CODE = 200 OK"
+		else
+			expected_result_="^4[0-9][0-9]"
+			expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+		fi
+		
+		actual_result="${http_code}"
+		if [[ "${http_code}" -eq "000" ]]; then
+			expected_log_msg="${rm_ml_error_messages[${exp}]}"
+			if grep -q "${expected_log_msg}" "${TESTS_}/service.log" ; then
+				failed=0 
+			else
+				failed=1
+			fi
+			error_message ${failed}
+		else
+			compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+		fi
+		echo ""
+		
+		# Stop the python servers
+		${SCRIPTS_DIR}/start_python_servers.sh -t | tee -a ${LOG_} ${LOG}
+		
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+		
+		echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
+		
+	done
+}
+
+# Do a post on experiment_trials for the same id again with "operation: EXP_TRIAL_GENERATE_NEW" and check if experiments have started from the begining
+function operation_generate_new() {
+	flag=0
+	get_json_array ${current_id}
+	post_json "${rm_ml_post_experiment_json[$exp]}"
+	
+	# Post the json with same Id having "operation: EXP_TRIAL_GENERATE_NEW"
+	echo "Post the json with same Id having operation: EXP_TRIAL_GENERATE_NEW" | tee -a ${LOG_} ${LOG}
+	post_json "${rm_ml_post_experiment_json[$exp]}"
+		
+	actual_result="${http_code}"
+	expected_result_="^4[0-9][0-9]"
+	expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+	
+	compare_result ${FUNCNAME} ${expected_result_} "${expected_behaviour}"
+}
+
+# Post the experiment result to python
+# input: Experiment result
+# output: Create the Curl command with given JSON and get the result
+function post_trial() {
+	exp_result='{"id" : "'${current_id}'", "trial_number": 0, "trial_result": "success", "result_value_type": "double", "result_value": 98.78, "operation" : "EXP_TRIAL_RESULT"}'
```

Comment:
Don't use 0 as the trial_number rather use the trial number returned from experiment_trials POST